feat(rel-notes): add performance improvements (#624)
bonitasoft/bonita-doc: md/release-notes.md

 - **BPM Services** page for the profile Technical Administrator
   It has been created using the UI Designer and the new modal container. Since it is dedicated to a profile critical for the good initialization and management of Bonita, the page has not been made available in the portal's resources. It cannot be added to any application nor be edited in the UI Designer. It's designed has been slightly reviewed compared to the GWT one.
   
<a id="performance"/>

### Performance

> A few improvements have been made to the engine efficiency:

Reviewer: ℹ️ I would not put emphasis on the engine part as even most of the improvements have been done on that component, Portal parts have also been improved (profiles and login)

Author: removed. thx

> A few improvements have been made to the engine efficiency:

Reviewer: ```suggestion
Improvements have been made in the following area:
```

Author: changed into "overall Bonita" efficiency

> A few improvements have been brought to the overall Bonita efficiency:

Reviewer: ℹ️ can we have the same wording in 7.7.5 and 7.8.0?

Author: OK. Copied and pasted.

> - Deleting an archived case now takes only a few milliseconds

Reviewer: ℹ️ seems maybe a too big commitment (few milliseconds) but we can say that deletion is done in a factor of milliseconds whereas in previous Bonita versions, this was done with a factor of seconds
We could also insist on the fact that
* this is particularly visible on environments with a large amount of archived data
* batch deletion
  * is almost as fast as single deletion. Previously the time increased in proportion of the number of the cases to be deleted
  * was then constrained by the overall timeout process (mainly database connection timeout) so its usage was limited only to environments with medium dataset
  * now provides more maintenance facilities because it is faster and better works on large environments

Author: made a summary

> - The number of accesses to temp folders has been reduced on the file system to limit I/O access 

Reviewer: ```suggestion
- There is less I/O requests to temp folders
```

Reviewer: I am very confident with the wording with may ask someone else

Author: I'm taking your words in the meantime

> - Deleting an archived case went down from seconds to milliseconds. Particularly enjoyable on environments with a large amount of archived data. Moreover, deleting cases in batches is now nearly as fast as deleting a single case and is not constrained by database connexion timeouts. 

Reviewer: ℹ️ we should change this as the improvement is only partial for now (see what we document for 7.7.5)

Author: ok. Changed.

- Works execution uses less database resources
- The login REST APIs mechanism consumes less resources

> - At engine startup and BPM services resume, jars are written on disk right after retrieving them from the database to save memory 

Reviewer: ℹ️ I would add the impacted use case in addition to technical explanation. I am not sure this is releavant to dig into so much technical details. I think that the result is more important
Let's talk about it directly

Author: There are many customers encountering the issues you have fixed. 
Non-technical people will not dig in but just see there are many areas of improvements.
Technical people will notice the potential amount of improvement, that we ourselves cannot easily quantify.

> - At engine startup and BPM services resume, jars are written on disk right after retrieving them from the database to save memory 

Reviewer: ```suggestion
- At engine startup and BPM services resume, the process dependencies loading requires less memory
```

Author: thx

> 

Reviewer: ⚠️ special warning about should be added about the triggered events change
See what we have in 7.7.5

Author: ok. Changed.

Reviewer: Looks ok now 😄
